src/IECoreScene/bindings/ShaderNetworkAlgoBinding.cpp

```diff
@@ -50,6 +50,11 @@ void convertOSLComponentConnectionsWrapper( ShaderNetwork *network, int oslVersi
 	ShaderNetworkAlgo::convertOSLComponentConnections( network, oslVersion );
 }
 
+CompoundDataPtr expandSplineParametersWrapper( CompoundDataPtr parameters )
+{
+	return ShaderNetworkAlgo::expandSplineParameters( parameters )->copy();
```

Comment:
The semantics of return-the-original-or-a-modified-copy on the C++ side are already a bit subtle, so I think matching them exactly in the Python bindings would be preferable to introducing another behaviour. Since everything in Python is non-const, we usually do that with `const_pointer_cast`.